internal/config/profile.go

```diff
@@ -27,9 +27,22 @@ import (
 	"github.com/spf13/viper"
 )
 
-//go:generate mockgen -destination=../mocks/mock_profile.go -package=mocks github.com/mongodb/mongocli/internal/config Setter,Saver,SetSaver,Getter,Config
+//go:generate mockgen -destination=../mocks/mock_profile.go -package=mocks github.com/mongodb/mongocli/internal/config SetSaver
```

Comment:
this is the main change cleaning unused interfaces and inverting how this work a bit to make it more in line with go practices (return structs accept interfaces)